### ISSUE TITLE
Update notifications not working as intended

### ISSUE THREAD

**Issue by Richard238** [MichMich/MagicMirror#2644]
From - https://forum.magicmirror.builders/topic/15425

`git remote -v`

origin https://github.com/MichMich/MagicMirror.git (fetch)
origin https://github.com/MichMich/MagicMirror.git (push)

config is set to get update notifications:

`{
 module: "updatenotification", position: "top_bar" 
}, `

I received no notification of the upgrade to 2.15, or to 2.16.

When I manually applied an update, I was some 4130 commits behind.

**Comment by rejas** [comment]
A quick check seems to indicate that the updatenotification worked until 2.15.0 
With that release something broke so that only module updates are shown but not the core updates.

**Comment by rejas** [comment]
Maybe @buxxi can take a look at it since he worked on that module during the developemnt cycle.

**Comment by khassel** [comment]
This is caused by https://github.com/MichMich/MagicMirror/pull/2100.

Running with `--dry-run` the property `fetchData.behind` is always = 0.

Running without `--dry-run` works as expected.

So if there is no better idea we have to revert the above PR.

**Comment by rejas** [comment]
So @bee-mar should weigh in on this topic :-)

**Comment by buxxi** [comment]
I didn't receive a notification for the last release, have been thinking about trying to debug the cause but @khassel seems to have been quicker.

I have one module that hasn't been updated that I tested an alternative way to find out the amount of commits:
```sh
git fetch --dry-run
```
> From https://github.com/timdows/MMM-JsonTable
>   8059e8d..3051433  master     -> origin/master
```sh
git rev-list --ancestry-path 8059e8d..3051433 | wc -l
```
> 4

haven't checked how this would work in the module, but it doesn't actually update the local git with the remote. Wouldn't want every module to auto update when a commit is pushed to the repo :)

**Comment by khassel** [comment]
This can be closed, fixed in new release `v2.17.1`.
